evalml/pipelines/time_series_regression_pipeline.py

```diff
@@ -86,7 +88,11 @@ def predict(self, X, y=None, objective=None):
         y = _convert_woodwork_types_wrapper(y.to_series())
 
         features = self.compute_estimator_features(X, y)
-        predictions = self.estimator.predict(features.dropna(axis=0, how="any"))
+
+        if 'y' in inspect.signature(self.estimator.predict).parameters:
```

Comment:
I see. So right now the baseline regressor for time series is the only one that takes `y` in the predict method. This is great but can you please add a comment explaining why we need the call to inspect? I wonder if there's some other mechanism we can use here?